### ISSUE TITLE
`Select` component: `SubmitEvent` triggered instead of `OnSelect` for ` <Select<String> ... variant={SelectVariant::Multiple(onchange_category)}>...`

### ISSUE THREAD

**Issue by collinsmarra** [patternfly-yew/patternfly-yew#63]
```rs
...
    let onchange_category = {
        let state = state.clone();
        use_callback(move|category: Vec<String>, state|{
            gloo::console::log!(category.clone().join(","));
            let mut data = state.deref().clone();
            data.categories = category.join(",");
            state.set(data)
        }, state)
    };
    
   html!{
....
                <FormSection>
                    <FormGroup label="Select Categories">
                        <Select<String> chip={ChipVariant::Values} variant={SelectVariant::Multiple(onchange_category)}>
                                {
                                    for categories_vec.iter().map(|category|{
                                        html_nested!{<SelectOption<String> value={category.to_string()}/>}
                                    })
                                }
                        </Select<String>>
                    </FormGroup>
                    ....
}
```

[Screencast from 06-26-2023 12:38:47 PM.webm](https://github.com/ctron/patternfly-yew/assets/48761741/3f8eab9c-e2a1-4584-b27b-8b7b9b68b590)

As you can see every time, an item is selected, it is treated as `SubmitEvent` and a request will be sent to the backend.

**Comment by ctron** [comment]
Yup, IIRC that was fixed just recently. One of the buttons was missing the `type="button"` attribute.

**Comment by ctron** [comment]
Ah now, same issue I guess, but different component: https://github.com/patternfly-yew/patternfly-yew/issues/42

**Comment by ctron** [comment]
There should be a fix on `main` (which uses PF5 now) or on `release-0.4.x` (for PF4).

**Comment by collinsmarra** [comment]
I have been migrating and I am having to change lots of things. I am not sure if `SubmitEvent`  not actually works.
For Example:

```rs
    let onsubmit = {
        let local_errors = local_errors;
        let (state, dispatch) = (login_state.clone(), dispatch);
        Callback::from(move |event: SubmitEvent| {
            event.prevent_default();
            let state = state.clone();
            let local_errors = local_errors.clone();
            let (email, password, dispatch) = (
                state.email.clone(),
                state.password.clone(),
                dispatch.clone(),
            );
            wasm_bindgen_futures::spawn_local(async move {
                let response = login(&email, &password).await;
                let local_errors = local_errors.clone();
                match response {
                    Ok(response) => {
                        // FIXME: set cookie
                        redirect_page("/pages/display".to_string());
                    }
                    Err(err) => {
                        // web_sys::console::log_1(&JsValue::from_str(&err.description.to_string()));
                            let mut data = local_errors.deref().clone();
                            data.message = err.message.clone();
                            data.description = err.description.clone();
                            local_errors.set(data);

                    }
                }
            })
        })
    };

    html! {
        <>
            <Background />
            {error_handler}
                <Login>
                    <LoginMain>
                        <LoginMainHeader title={title} description="Enter Credentials to login to your account"/>
                            <LoginMainBody>
                                <Form onsubmit={onsubmit}>
                                    <FormGroup label="Email">
                                        <TextInput 
                                            required=true name="email" r#type={TextInputType::Email}
                                            value={(*login_state.email).to_string().clone()}
                                            onchange={oninput_email} />
                                    </FormGroup>
                                    <FormGroup label="Password">
                                        <TextInput 
                                            required=true name="password" r#type={TextInputType::Password} 
                                            value={(*login_state.password).to_string().clone()}
                                            onchange={oninput_password}/>
                                    </FormGroup>
                                    <ActionGroup>
                                        <Button label="log in" r#type={ButtonType::Submit} variant={ButtonVariant::Primary} />
                                    </ActionGroup>
                                </Form>
                            </LoginMainBody>
                        <LoginMainFooter links={links} ></LoginMainFooter>
                    </LoginMain>
                </Login>
        </>
    }
  ```

**Comment by phill-85** [comment]
> I have been migrating and I am having to change lots of things.

Have you seen this [document](https://github.com/patternfly-yew/patternfly-yew/blob/main/docs/MIGRATION-0_5.md)? Feel free to add or file an issue for anything you think might be missing!

**Comment by collinsmarra** [comment]
@phill-85 thanks, but I managed to migrate most of the things, there is just an issue with `OnSubmit`

**Comment by ctron** [comment]
> there is just an issue with `OnSubmit`

What's the issue?

**Comment by collinsmarra** [comment]
when the "log in" button is clicked,  the onsubmit callback is not triggered. I could  be missing something, but I have checked and rechecked 

> I have been migrating and I am having to change lots of things. I am not sure if `SubmitEvent` not actually works. For Example:
> 
> ```rust
>     let onsubmit = {
>         let local_errors = local_errors;
>         let (state, dispatch) = (login_state.clone(), dispatch);
>         Callback::from(move |event: SubmitEvent| {
>             event.prevent_default();
>             let state = state.clone();
>             let local_errors = local_errors.clone();
>             let (email, password, dispatch) = (
>                 state.email.clone(),
>                 state.password.clone(),
>                 dispatch.clone(),
>             );
>             wasm_bindgen_futures::spawn_local(async move {
>                 let response = login(&email, &password).await;
>                 let local_errors = local_errors.clone();
>                 match response {
>                     Ok(response) => {
>                         // FIXME: set cookie
>                         redirect_page("/pages/display".to_string());
>                     }
>                     Err(err) => {
>                         // web_sys::console::log_1(&JsValue::from_str(&err.description.to_string()));
>                             let mut data = local_errors.deref().clone();
>                             data.message = err.message.clone();
>                             data.description = err.description.clone();
>                             local_errors.set(data);
> 
>                     }
>                 }
>             })
>         })
>     };
> 
>     html! {
>         <>
>             <Background />
>             {error_handler}
>                 <Login>
>                     <LoginMain>
>                         <LoginMainHeader title={title} description="Enter Credentials to login to your account"/>
>                             <LoginMainBody>
>                                 <Form onsubmit={onsubmit}>
>                                     <FormGroup label="Email">
>                                         <TextInput 
>                                             required=true name="email" r#type={TextInputType::Email}
>                                             value={(*login_state.email).to_string().clone()}
>                                             onchange={oninput_email} />
>                                     </FormGroup>
>                                     <FormGroup label="Password">
>                                         <TextInput 
>                                             required=true name="password" r#type={TextInputType::Password} 
>                                             value={(*login_state.password).to_string().clone()}
>                                             onchange={oninput_password}/>
>                                     </FormGroup>
>                                     <ActionGroup>
>                                         <Button label="log in" r#type={ButtonType::Submit} variant={ButtonVariant::Primary} />
>                                     </ActionGroup>
>                                 </Form>
>                             </LoginMainBody>
>                         <LoginMainFooter links={links} ></LoginMainFooter>
>                     </LoginMain>
>                 </Login>
>         </>
>     }
> ```

**Comment by collinsmarra** [comment]
> when the "log in" button is clicked, the onsubmit callback is not triggered. I could be missing something, but I have checked and rechecked
> 
> > I have been migrating and I am having to change lots of things. I am not sure if `SubmitEvent` not actually works. For Example:
> > ```rust
> >     let onsubmit = {
> >         let local_errors = local_errors;
> >         let (state, dispatch) = (login_state.clone(), dispatch);
> >         Callback::from(move |event: SubmitEvent| {
> >             event.prevent_default();
> >             let state = state.clone();
> >             let local_errors = local_errors.clone();
> >             let (email, password, dispatch) = (
> >                 state.email.clone(),
> >                 state.password.clone(),
> >                 dispatch.clone(),
> >             );
> >             wasm_bindgen_futures::spawn_local(async move {
> >                 let response = login(&email, &password).await;
> >                 let local_errors = local_errors.clone();
> >                 match response {
> >                     Ok(response) => {
> >                         // FIXME: set cookie
> >                         redirect_page("/pages/display".to_string());
> >                     }
> >                     Err(err) => {
> >                         // web_sys::console::log_1(&JsValue::from_str(&err.description.to_string()));
> >                             let mut data = local_errors.deref().clone();
> >                             data.message = err.message.clone();
> >                             data.description = err.description.clone();
> >                             local_errors.set(data);
> > 
> >                     }
> >                 }
> >             })
> >         })
> >     };
> > 
> >     html! {
> >         <>
> >             <Background />
> >             {error_handler}
> >                 <Login>
> >                     <LoginMain>
> >                         <LoginMainHeader title={title} description="Enter Credentials to login to your account"/>
> >                             <LoginMainBody>
> >                                 <Form onsubmit={onsubmit}>
> >                                     <FormGroup label="Email">
> >                                         <TextInput 
> >                                             required=true name="email" r#type={TextInputType::Email}
> >                                             value={(*login_state.email).to_string().clone()}
> >                                             onchange={oninput_email} />
> >                                     </FormGroup>
> >                                     <FormGroup label="Password">
> >                                         <TextInput 
> >                                             required=true name="password" r#type={TextInputType::Password} 
> >                                             value={(*login_state.password).to_string().clone()}
> >                                             onchange={oninput_password}/>
> >                                     </FormGroup>
> >                                     <ActionGroup>
> >                                         <Button label="log in" r#type={ButtonType::Submit} variant={ButtonVariant::Primary} />
> >                                     </ActionGroup>
> >                                 </Form>
> >                             </LoginMainBody>
> >                         <LoginMainFooter links={links} ></LoginMainFooter>
> >                     </LoginMain>
> >                 </Login>
> >         </>
> >     }
> > ```



> > there is just an issue with `OnSubmit`
> 
> What's the issue?

**Comment by ctron** [comment]
I think I can reproduce the issue. Need to check the cause and fix.

**Comment by ctron** [comment]
Yea, I think I know what the issue is. We started to use `AttrValue`, as recommended by Yew. However using `AttrValue` with a default will actually create an attribute with an empty string. Which caused some problems already, especially with the `form` attribute. Which then points to a form with an empty ID. Which doesn't work.

Switching to `Option<AttrValue>` solved this issue. And I guess we need to check all the places for exactly that.

**Comment by ctron** [comment]
This should be fixed with c33dcb697fbf6d8217db30294a849f607d743629
